File: examples/cpp/RVC4/Audio/audio_list.cpp
#include <iostream>

#include "depthai/depthai.hpp"
#include "depthai/utility/AudioHelpers.hpp"

int main() {
    dai::Device device;

    auto devices = device.getAlsaDevices();
    for(auto device : devices) {
        std::cout << "Name: " << device.name << std::endl << "Desc: " << device.desc << std::endl << "IOID: " << device.ioid << std::endl;
    }

    auto pcms = device.getAlsaPCMs();
    for(auto pcm : pcms) {
        std::cout << "Name: " << pcm.name << std::endl
                  << "ID: " << pcm.id << std::endl
                  << "Card #: " << pcm.cardNumber << std::endl
                  << "Device #: " << pcm.deviceNumber << std::endl;
    }
}

File: examples/python/Audio/audio_list.py
#!/usr/bin/env python3

import depthai as dai

# Create pipeline
device = dai.Device()

devices = device.getAlsaDevices();

for d in devices:
    print("Name: " + d.name + "\n");
    print("Desc: " + d.desc + "\n");
    print("IOID: " + d.ioid + "\n");

pcms = device.getAlsaPCMs();

for p in pcms:
    print("Name:  " + p.name + "\n");
    print("ID:    " + p.id+ "\n");
    print("Card#: " + p.cardNumber + "\n");
    print("Dev#:  " + p.deviceNumer + "\n");

